src/scoring/hpf_scoring.py

```diff
@@ -159,130 +113,102 @@ def predict(self, input_stack):
-        rec = np.dot(theta, self.beta.T)
-        return self.filter_recommendation(rec, input_user_stack)
-
-    def filter_recommendation(self, result, input_stack):
-        """Filter based on feedback, input set & max recommendation count.
-
-        :param result: The unfiltered companion recommendation result.
-        :return companion_recommendation: The filtered list of recommended companion packages
-        along with condifence score.
+        Implement the 'add_user' addon of HPFREC model.
+
+        For more information please follow:
+        https://hpfrec.readthedocs.io/en/latest/source/hpfrec.html
         """
-        companion_recomendations = []
-        input_stack = list(input_stack)
-        result = np.delete(result, input_stack)
-        result = np.where(np.isfinite(result))[0]
-        if self.USE_FEEDBACK:
-            alpha_id = int(self.match_feedback_manifest(set(input_stack)))
-            alpha_set = set(np.where(self.alpha[alpha_id] >= feedback_threshold)[0])
-            result = np.delete(result, list(alpha_set))
-
-        def _sigmoid(x):
-            return 1 / (1 + np.exp(-x))
-        recommended_ids = result.argsort()[::-1][:MAX_COMPANION_REC_COUNT]
-        mean = np.mean(result[recommended_ids])
-        result = result - mean
-        result = np.divide(result, np.std(result))
-        for package_id in recommended_ids:
-            recommendation = {
-                "cooccurrence_probability": round(_sigmoid(result[package_id]) * 100, 2),
-                "package_name": self.id_package_dict[package_id],
-                "topic_list": []  # We don't have topics for this ecosystem!!
-            }
-            # At least thirty percent probability is required for recommendation to go through.
-            if recommendation['cooccurrence_probability'] > MIN_REC_CONFIDENCE:
-                companion_recomendations.append(recommendation)
-        return companion_recomendations
+        # no blank line allowed here
+        new_df = pd.DataFrame({
+            'ItemId': input_id_set,
+            'Count': [1] * len(input_id_set)})
+        user_id = self.recommender.nusers
+        is_user_added = self.recommender.add_user(user_id=user_id, counts_df=new_df)
+        if is_user_added:
+            user_id -= 1
+            recommendations = self.recommender.topN(user_id, n=self.m)
+            return recommendations, user_id
+
+        raise ValueError("Unable to add user")
+
+    def filter_recommendation(self, result, input_stack, user_id):
+        """Use for co-occurrence probability and for filtering of companion packages."""
+        package_id_set = input_stack
+        recommendations = result
+        companion_packages = []
+        recommendations = \
+            np.array(list(itertools.compress(recommendations,
+                                             [i not in package_id_set for i in recommendations])))
+
+        print("Filtered recommendation ids are: " + str(recommendations))
```

Comment:
Don't use print statements. Use logging instead.

Reply:
Removed print statement, was using for debugging !